.github/workflows/test-load-k6.yaml

```
@@ -0,0 +1,64 @@
+name: k6 Load Test
+
+on:
+  push:
+    branches:
+      - "main"
+      - "rc"
+      - "hotfix-rc"
+
+jobs:
+  run-tests:
+    name: Run k6 tests
+    runs-on: ubuntu-24.04
+    services:
+      # Datadog agent for collecting OTEL metrics from k6
+      datadog:
+        image: datadog/agent:7-full
+        ports:
+          - 4318:4318
+          - 5555:5555
+        env:
+          DD_SITE: us3.datadoghq.com
+          DD_API_KEY: ${{ secrets.DD_API_KEY }}
+          DD_DOGSTATSD_NON_LOCAL_TRAFFIC: 1
+          DD_OTLP_CONFIG_RECEIVER_PROTOCOLS_HTTP_ENDPOINT: 0.0.0.0:4318
+          DD_HEALTH_PORT: 5555
+          HOST_PROC: /proc
+        options: >-
+          --volume /var/run/docker.sock:/var/run/docker.sock:ro
+          --volume /sys/fs/cgroup/:/host/sys/fs/cgroup:ro
+          --health-cmd "curl -f http://localhost:5555/health || exit 1"
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 10
+          --health-start-period 30s
+          --pid host
+    env:
+      # K6 and Datadog system configuration
+      _DOCKER_CONTENT_TRUST: 1
+      _K6_OTEL_EXPORTER_TYPE: http
+      _K6_OTEL_HTTP_EXPORTER_INSECURE: true
+      _K6_OTEL_METRIC_PREFIX: k6_
+      _K6_FLAGS: >-
+        --tag test-id=${{ github.ref_name }}
+        -o experimental-opentelemetry
+      
+      # Paths to your k6 load tests
+      _K6_LOAD_TESTS_PATHS: >-
+        perf/load/*.js
+    steps:
+      - name: Check out repo
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          persist-credentials: false
+
+      - name: Setup k6
```

```suggestion
      - name: Set up k6
```
At this step level I think it's reasonable to mention k6 though.